### PR TITLE
More robust building of polygon geometries from bboxes

### PR DESCRIPTION
**Pull request by SlowMo24**
OSHDBGeometryBuilder was returning empty Geometry for a point-like Bounding-Box. Now we force rectangle-like polygons.

### CODE REVIEW

File: oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java [hunk 1]
```diff
@@ -279,24 +279,31 @@ private static Geometry getMultiPolygonGeometry(
       T entity, OSHDBTimestamp timestamp, TagInterpreter areaDecider, P clipPoly) {
     Geometry geom = OSHDBGeometryBuilder.getGeometry(entity, timestamp, areaDecider);
     return Geo.clip(geom, clipPoly);
+  }
```
Comment (tyrasd) [inline oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java]: it looks like there is a superfluous space after the `}` here.

File: oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java [hunk 2]
```diff
@@ -279,24 +279,31 @@ private static Geometry getMultiPolygonGeometry(
...
+   * Create the Geometry of an OSHDBBoundingBox. Will always contain 5 Coordinates, even for
+   * point-like bbx -> point-like polygons for bbx of a Point.
+   *
+   * @param bbox The BoundingBox the polygon should be created for.
+   * @return com.vividsolutions.jts.geom.Geometry Returns JTS geometry object for convenience.
+   * Result is an empty polygon for a null BBX.
+   */
+  public static Polygon getGeometry(OSHDBBoundingBox bbox) {
+    GeometryFactory gf = new GeometryFactory();
+    if (bbox == null) {
```
Comment (tyrasd) [inline oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java]: why the extra check against `null` here? As far as I know, we never generate null bounding boxes.
Maybe add an assert instead (`assert bbox != null : "a bounding box is not allowed to be null";` or so)?

File: oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java [hunk 3]
```diff
@@ -279,24 +279,31 @@ private static Geometry getMultiPolygonGeometry(
...
     return Geo.clip(geom, clipPoly);
+  } 
+  
+  /**
+   * Create the Geometry of an OSHDBBoundingBox. Will always contain 5 Coordinates, even for
+   * point-like bbx -> point-like polygons for bbx of a Point.
+   *
+   * @param bbox The BoundingBox the polygon should be created for.
+   * @return com.vividsolutions.jts.geom.Geometry Returns JTS geometry object for convenience.
+   * Result is an empty polygon for a null BBX.
```
Comment (tyrasd) [inline oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java]: i think this should be indented according to our style guide. also, I think we never use the abbreviation _BBX_ for bounding boxes, but rather use _bbox_ if we want to have something short…

File: oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java [hunk 4]
```diff
@@ -279,24 +279,31 @@ private static Geometry getMultiPolygonGeometry(
       T entity, OSHDBTimestamp timestamp, TagInterpreter areaDecider, P clipPoly) {
     Geometry geom = OSHDBGeometryBuilder.getGeometry(entity, timestamp, areaDecider);
     return Geo.clip(geom, clipPoly);
+  } 
+  
+  /**
+   * Create the Geometry of an OSHDBBoundingBox. Will always contain 5 Coordinates, even for
+   * point-like bbx -> point-like polygons for bbx of a Point.
```
Comment (tyrasd) [inline oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java]: I think we could write that the result is always a rectangular _polygon_ with exactly 4 vertices. (5 coordinate geometries could be many things :sweat_smile: )

File: oshdb-util/src/test/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilderTest.java [hunk 5]
```diff
@@ -160,6 +162,28 @@ public void testRelationGetGeometryIncomplete() {
     assertEquals("GeometryCollection", result.getGeometryType());
     assertEquals(1, result.getNumGeometries());
   }
+  
+  @Test
+  public void testBoundingBoxGetGeometry() {
+    OSHDBBoundingBox bbx = new OSHDBBoundingBox(0, 0, 0, 0);
```
Comment (tyrasd) [inline oshdb-util/src/test/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilderTest.java]: maybe better start the test with the "regular" case, and then test the more specific corner cases like point/line degenerated bbox? also a comment line would be nice here to explain why these need to be tested differently.

File: oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java (hunk 3 above)
Comment (tyrasd) [reply oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java]: oh, and is there a reason to mention the class of the result object here in the `@return` javadoc? I mean, doesn't javadoc see that from the definition of the function itself already?

File: oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java (hunk 1 above)
Comment (SlowMo24) [reply oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java]: sorry for all the whytespace errors the last time. I need to get my IDE running with this...

File: oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java (hunk 2 above)
Comment (SlowMo24) [reply oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java]: yes I thought of that too. Will implement...

File: oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java (hunk 3 above)
Comment (SlowMo24) [reply oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java]: just copied the javadoc that was in the header before. Was wondering, too :-)

File: oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java (hunk 4 above)
Comment (SlowMo24) [reply oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java]: well... the check for geometry.isRectangular() does not always return true (eg. if the bbox has the form of a line). But one of the first checks isRect.. does is _does it have exactly 5 coordinates) so I used this more generic description

File: oshdb-util/src/test/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilderTest.java (hunk 5 above)
Comment (SlowMo24) [reply oshdb-util/src/test/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilderTest.java]: code-quality is really high on the agenda these days. I like it :+1: !!! Many things to learn for me.

File: oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java (hunk 4 above)
Comment (SlowMo24) [reply oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/geometry/OSHDBGeometryBuilder.java]: clarified in javadoc
